src/runtime-rs/src/shim/src/event_publisher.rs

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) 2021 Alibaba Cloud
+// Copyright (c) 2021 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::io::Write;
+use std::process::{Command, Stdio};
+use std::sync::mpsc::{channel, Receiver, Sender};
+use std::sync::{Arc, Mutex};
+
+use anyhow::{Context, Result};
+use log::{debug, error, info, warn};
+use protobuf::{well_known_types::Any, Message};
+
+use agent::Agent;
+use shim_proto::events_task::TaskOOM;
+
+const TASK_OOMEVENT_TOPIC: &str = "/tasks/oom";
+
+pub trait Event {
+    fn r#type(&self) -> String;
+    fn type_url(&self) -> String;
+}
+
+impl Event for TaskOOM {
+    fn r#type(&self) -> String {
+        TASK_OOMEVENT_TOPIC.to_string()
+    }
+
+    fn type_url(&self) -> String {
+        "containerd.events.TaskOOM".to_string()
```

Comment:
What is this string exactly? Is it going to be valid even if the container manager is not containerd?

Reply:
Fixed in the first revision